js/main.js

```diff
@@ -34,3 +35,77 @@ if (previousData !== null) {
 window.addEventListener('beforeunload', function () {
   localStorage.setItem('UserData', JSON.stringify(currentData));
```

Comment:
You shouldn't use `currentData` here, since that is the data from when the page first loads. You want to save the most up-to-date data from the `data` variable 👍